code/game/objects/items/weapons/explosives.dm

```diff
@@ -68,16 +71,19 @@
 
 		target.overlays += image_overlay
 		to_chat(user, "Bomb has been planted. Timer counting down from [timer].")
-		spawn(timer*10)
-			explode(get_turf(target))
+		addtimer(CALLBACK(src, .proc/detonate), timer SECONDS)
 
-/obj/item/weapon/plastique/proc/explode(var/location)
+/obj/item/weapon/plastique/proc/detonate()
 	if(!target)
 		target = get_atom_on_turf(src)
 	if(!target)
 		target = src
-	if(location)
-		explosion(location, blast_dev, blast_heavy, blast_light, blast_flash)
+	var/turf/T = get_turf(src)
+	if(!istype(T) && location.len == 3)

```

Comment:
General: prefer core procs (isturf) where available for those teeny tiny gains.

Not sure this is an ideal behavior since it makes the C4 able to detonate from null loc among other things. It would surely also be preferable to give up on other stuff and always use the x/y/z either way, if using this approach?